legend-engine-xts-service/legend-engine-language-pure-dsl-service/src/main/java/org/finos/legend/engine/language/pure/dsl/service/grammar/from/ServiceParseTreeWalker.java

```diff
@@ -575,6 +575,10 @@ private SingleExecutionParameters visitSingleExecutionParameters(ServiceParserGr
         SingleExecutionParameters singleExecParams = new SingleExecutionParameters();
         singleExecParams.sourceInformation = walkerSourceInformation.getSourceInformation(ctx);
         singleExecParams.key = PureGrammarParserUtility.fromIdentifier(ctx.identifier());
+        if (singleExecParams.key.contains(" "))
+        {
+            throw new EngineException("Execution param key cannot contain spaces. Invalid Key: " + singleExecParams.key);
```

Comment:
we should include Exception type and source info so error is highllghted correctly to the user.